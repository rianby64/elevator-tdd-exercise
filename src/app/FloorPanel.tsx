
import * as React from 'react';

interface FloorPanelProps {
    upButton: boolean;
    downButton: boolean;
    callLift?: () => void;
    highlight?: boolean;
}

type FloorPanelElement = React.ReactElement<FloorPanelProps>;

export const FloorPanel = (props: FloorPanelProps): FloorPanelElement => {
    if (!props.upButton && !props.downButton) {
        return (<div></div>);
    }
    return (
        <ul className={props.highlight ? "highlight" : undefined}>
            {props.upButton && (<button onClick={props.callLift}>Up</button>)}
            {props.downButton && (<button onClick={props.callLift}>Down</button>)}
        </ul>
    );
}
