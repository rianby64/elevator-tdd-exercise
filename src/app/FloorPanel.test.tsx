
import * as React from 'react';
import { mount } from 'enzyme';
import { FloorPanel } from './FloorPanel';

test('FloorPanel has up and down buttons', (): void => {
    const el = (<FloorPanel upButton={true} downButton={true} />);
    const sh = mount(el);
    expect(sh).toMatchSnapshot();
});

test('FloorPanel has only up button', (): void => {
    const el = (<FloorPanel upButton={true} downButton={false} />);
    const sh = mount(el);
    expect(sh).toMatchSnapshot();
});

test('FloorPanel has down button', (): void => {
    const el = (<FloorPanel upButton={false} downButton={true} />);
    const sh = mount(el);
    expect(sh).toMatchSnapshot();
});

test('FloorPanel has no buttons', (): void => {
    const el = (<FloorPanel upButton={false} downButton={false} />);
    const sh = mount(el);
    expect(sh).toMatchSnapshot();
});

test('FloorPanel allows to call the lift', (): void => {
    let liftCalled = false;
    const callLift = (): void => {
        liftCalled = true;
    }
    const el = (<FloorPanel
        upButton={true} downButton={false}
        callLift={callLift} />);
    const sh = mount(el);
    sh.find('li').simulate('click');
    expect(liftCalled).toBe(true);
});
