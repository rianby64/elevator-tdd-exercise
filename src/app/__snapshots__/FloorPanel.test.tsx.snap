// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloorPanel has down button 1`] = `
<Component
  downButton={true}
  upButton={false}
>
  <ul>
    <button>
      Down
    </button>
  </ul>
</Component>
`;

exports[`FloorPanel has no buttons 1`] = `
<Component
  downButton={false}
  upButton={false}
>
  <div />
</Component>
`;

exports[`FloorPanel has only up button 1`] = `
<Component
  downButton={false}
  upButton={true}
>
  <ul>
    <button>
      Up
    </button>
  </ul>
</Component>
`;

exports[`FloorPanel has up and down buttons 1`] = `
<Component
  downButton={true}
  upButton={true}
>
  <ul>
    <button>
      Up
    </button>
    <button>
      Down
    </button>
  </ul>
</Component>
`;
